### PR TITLE
Feature/project group

### PR DESCRIPTION
**Pull request by jemrayfieldft**
Added Variant and Create Place Holder Group action

### CODE REVIEW

File: spec/factories/model_factories.rb [hunk 1]
```diff
@@ -39,15 +39,16 @@
...
-    visibility [4]
-    enable :true
 
     factory :s3 do
       name 'S3 Bucket Factory'
       key 'some_s3_key'
       secret 'some_s3_secret'
       bucket 'some_faked_s3_bucket'
       start :true
+      enable :true
```
Comment (jspc) [inline spec/factories/model_factories.rb]: This is awful- at least make it a trait

File: lib/mio/model/create_place_holder_group_asset.rb [hunk 2]
```diff
@@ -0,0 +1,51 @@
+class Mio
+  class Model
+    class CreatePlaceHolderGroupAsset < Model
+      set_resource :actions
```
Comment (jspc) [inline lib/mio/model/create_place_holder_group_asset.rb]: Is this right?

File: lib/mio/model/create_place_holder_group_asset.rb [hunk 3]
```diff
@@ -0,0 +1,51 @@
+class Mio
+  class Model
+    class CreatePlaceHolderGroupAsset < Model
```
Comment (jspc) [inline lib/mio/model/create_place_holder_group_asset.rb]: This is a model called a 'CreatePlaceHolderAsset'? Thats not right, is it?

File: spec/lib/mio/model/metadatadefinition_spec.rb [hunk 4]
```diff
@@ -34,4 +34,4 @@
     end
   end
 
-end
\ No newline at end of file
+end
```
Comment (jspc) [inline spec/lib/mio/model/metadatadefinition_spec.rb]: :+1:

File: lib/mio/model/create_place_holder_group_asset.rb (hunk 2 above)
Comment (jemrayfieldft) [inline lib/mio/model/create_place_holder_group_asset.rb]: Is what right?

File: lib/mio/model/create_place_holder_group_asset.rb (hunk 3 above)
Comment (jemrayfieldft) [inline lib/mio/model/create_place_holder_group_asset.rb]: Yes it is...

We need an Action to create a place holder asset.

Comment (jspc) [inline lib/mio/model/create_place_holder_group_asset.rb]: So the action is create and the model is a PlaceHolderAsset

Comment (jemrayfieldft) [inline lib/mio/model/create_place_holder_group_asset.rb]: Plugin : 
tv.nativ.mio.enterprise.execution.action.file.process.impl.createobject.CreatePlaceholderAsset

File: lib/mio/model/create_place_holder_group_asset.rb (hunk 2 above)
Comment (jspc) [inline lib/mio/model/create_place_holder_group_asset.rb]: The resource name being actions- is it on `/api/actions`, with this creating an action?

File: lib/mio/model/create_place_holder_group_asset.rb (hunk 3 above)
Comment (jspc) [inline lib/mio/model/create_place_holder_group_asset.rb]: So thats the action, and the model is a PlaceholderAsset- that underlying plugin class name could be anything in mio.

File: lib/mio/model/create_place_holder_group_asset.rb (hunk 2 above)
Comment (jemrayfieldft) [inline lib/mio/model/create_place_holder_group_asset.rb]: Yes it is an action to create a placeholderasset.
/api/actions

File: lib/mio/model/create_place_holder_group_asset.rb (hunk 3 above)
Comment (jemrayfieldft) [inline lib/mio/model/create_place_holder_group_asset.rb]: ?? Sorry I dont get you ??

We need to create a

Place Holder Group Asset via an Action.

Thus the model.

Comment (jemrayfieldft) [inline lib/mio/model/create_place_holder_group_asset.rb]: The place holder will be associated to a project....after which the placeholder is updated with the metadata definition values...

Thats how mio rolls..

Comment (jemrayfieldft) [inline lib/mio/model/create_place_holder_group_asset.rb]: Are you suggesting that we open up plugin as a Field?

The issue is the configuration is specific to the plugin, right?

Thus a placeholder asset requires 
1) Variant
2) Metadata definition

See the other models
1) Variant
2) MetadataDefinition

Cheers
Jem

Comment (jemrayfieldft) [inline lib/mio/model/create_place_holder_group_asset.rb]: I can rename it if helps

CreateAssetGroupPlaceHolderAction

OK?

File: lib/mio/model/create_place_holder_group_asset.rb [hunk 5]
```diff
@@ -1,6 +1,6 @@
 class Mio
   class Model
-    class CreatePlaceHolderGroupAsset < Model
+    class CreatePlaceHolderGroupAssetAction < Model
```
Comment (jspc) [inline lib/mio/model/create_place_holder_group_asset.rb]: The action is create, the model is not- I'm not merging this, that naming is bad.

**Pull request merged**